http/http.go

```diff
@@ -66,6 +120,16 @@ func PostHelper(ctx context.Context, httpClient *http.Client, stats *statsd.Clie
 
 	req, err := http.NewRequest(http.MethodPost, endpoint, &bodyBuffer)
 
+	hct := NewHTTPClientTracer(stats, action)
+	httpTracer := &httptrace.ClientTrace{
+		GotConn:      hct.GotConn,
+		DNSStart:     hct.DNSStart,
+		DNSDone:      hct.DNSDone,
+		ConnectStart: hct.ConnectStart,
+		ConnectDone:  hct.ConnectDone,
+		WroteRequest: hct.WroteRequest,
+	}
+
```

Comment:
Can you just generate a pre-filled `httpTracer` from `NewHTTPClientTracer`? Seems unergonomic doing this here. You'll have to return both the `hct` and the `httpTracer` from that method (:

Reply:
Also, maybe `defer hct.FinishedRequest()` here once you do?